### DISCUSSION TITLE
Media not being removed from B2 bucket

### DISCUSSION THREAD

**Discussion by gill6151** [mastodon/mastodon#27133] [Q&A]
Hello all, 

When trying to run `tootctl media remove --days=2`, it appears to runs fine and reports back that a large amount of media is being removed, but the storage usage and file count according to Backblaze on my B2 bucket does not go down at all. 

According to the Mastodon administration panel and `tootctl media usage`, the media storage space is going down, but this is not reflected on the bucket itself which continues showing the same size as before as if no files have been deleted (or higher, if the process takes some time and newer media is pulled in the meantime). As of right now, Mastodon is reporting 18.8GB of media storage usage and Backblaze is reporting the media bucket size as 59.3GB.

When going in and viewing my buckets files, I can see that some files are being marked as having multiple versions, one of which being a "hidden" version instead of just removing the file completely, which contradicts what is supposed to be happening as my buckets lifecycle option is set to "Keep the latest version of every file" which as I understand it means these files should just be getting removed once there is an updated "hidden" version. 

It is unclear to me whether this is a Mastodon or Backblaze problem, any guidance is appreciated.

**Comment by StrangeMinds1960** [comment]
If it's anything like S3 buckets, it takes time (sometimes several hours) to remove all media marked for removal

**Reply by gill6151** [reply]
I've been watching this over the course of a few days now to no avail. 

Upon further research, with the way I have my bucket setup, content marked as hidden should be getting removed from the bucket after 24 hours, this is not happening by the looks so I am going to attribute this to more of a Backblaze issue unless someone else has experienced a similar thing and has a solution.

**Comment by gill6151** [comment]
Issue appears to have been resolved by Backblaze support, auto removal of hidden files was not functioning as it should.
